spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/utils/DistributionAndOrderingUtils.scala

```diff
@@ -118,7 +120,24 @@ object DistributionAndOrderingUtils {
     expr match {
       case s: SortOrder =>
         val catalystChild = toCatalyst(s.expression(), query, resolver)
-        catalyst.expressions.SortOrder(catalystChild, toCatalyst(s.direction), toCatalyst(s.nullOrdering), Set.empty)
+        val ctor: DynConstructors.Ctor[catalyst.expressions.SortOrder] =
```

Comment:
This constructor should be moved into a companion object so that the lookups don't happen every time the method is called.

Reply:
Moved the val defining the constructor from a local val to a val in the object.